src/server/serveractiveobjectmap.cpp

```diff
@@ -151,28 +159,38 @@ std::unordered_set<u16> ServerActiveObjectMap::getObjectsNearBox(const aabb3s16
 	return result;
 }
 
+void ServerActiveObjectMap::addObjectRef(u16 id, v3s16 pos)
+{
+	refmap.emplace(pos, id);
+}
+
+void ServerActiveObjectMap::removeObjectRef(u16 id, v3s16 pos)
+{
+	auto bounds = refmap.equal_range(pos);
+	for (auto iter = bounds.first; iter != bounds.second;) {
```

Comment:
why not just use std::remove_if ?

Reply:
@nerzhul *`remove_if` does not work for a map. (Neither for a set).*

Reply:
lol, nerzhul asked the exact same thing on the original PR: https://github.com/minetest/minetest/pull/6587#discussion_r148971940

Reply:
i ask too many things :p

Reply:
BTW, there is some code on SO (https://stackoverflow.com/questions/800955/remove-if-equivalent-for-stdmap):
```c++
  template< typename ContainerT, typename PredicateT >
  void erase_if( ContainerT& items, const PredicateT& predicate ) {
    for( auto it = items.begin(); it != items.end(); ) {
      if( predicate(*it) ) it = items.erase(it);
      else ++it;
    }
  };
```
The very reason of its existence is that `remove_if` works for sequences only, like ranges in a list, a vector, etc. Even there, it doesn’t perform well, e.g. on `std::list`, it can’t use its capability to remove elements from the middle, thus using tons of movement, more like `std::vector::pop_front`.